File: flutter/lib/models/audit_log.dart
class AuditLog {
  final int id;
  final int? userId;
  final int? employeeId;
  final String? role;
  final String action;
  final String endpoint;
  final String method;
  final int statusCode;
  final String ipAddress;
  final String userAgent;
  final String? requestBody;
  final String? responseBody;
  final int durationMs;
  final String? errorMessage;
  final DateTime createdAt;
  final Map<String, dynamic>? user;
  final Map<String, dynamic>? employee;

  AuditLog({
    required this.id,
    this.userId,
    this.employeeId,
    this.role,
    required this.action,
    required this.endpoint,
    required this.method,
    required this.statusCode,
    required this.ipAddress,
    required this.userAgent,
    this.requestBody,
    this.responseBody,
    required this.durationMs,
    this.errorMessage,
    required this.createdAt,
    this.user,
    this.employee,
  });

  factory AuditLog.fromJson(Map<String, dynamic> json) {
    DateTime parseDate(String key) {
      final value = json[key];
      if (value is String) {
        final parsed = DateTime.tryParse(value);
        if (parsed != null) return parsed;
      }
      return DateTime.now();
    }

    return AuditLog(
      id: json['id'] as int,
      userId: json['user_id'] as int?,
      employeeId: json['employee_id'] as int?,
      role: json['role'] as String?,
      action: json['action'] as String,
      endpoint: json['endpoint'] as String,
      method: json['method'] as String,
      statusCode: json['status_code'] as int,
      ipAddress: json['ip_address'] as String? ?? '',
      userAgent: json['user_agent'] as String? ?? '',
      requestBody: json['request_body'] as String?,
      responseBody: json['response_body'] as String?,
      durationMs: json['duration_ms'] as int? ?? 0,
      errorMessage: json['error_message'] as String?,
      createdAt: parseDate('created_at'),
      user: json['user'] as Map<String, dynamic>?,
      employee: json['employee'] as Map<String, dynamic>?,
    );
  }

  String get statusEmoji {
    if (statusCode >= 200 && statusCode < 300) return '✅';
    if (statusCode >= 300 && statusCode < 400) return '🔄';
    if (statusCode >= 400 && statusCode < 500) return '⚠️';
    if (statusCode >= 500) return '❌';
    return '❓';
  }

  String get methodColor {
    switch (method.toUpperCase()) {
      case 'GET':
        return 'blue';
      case 'POST':
        return 'green';
      case 'PUT':
      case 'PATCH':
        return 'orange';
      case 'DELETE':
        return 'red';
      default:
        return 'grey';
    }
  }
}

File: golang/models/audit_log.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type AuditLog struct {
	ID           uint           `gorm:"primaryKey" json:"id"`
	UserID       *uint          `gorm:"index" json:"user_id,omitempty"`
	User         *User          `gorm:"foreignKey:UserID;references:ID" json:"user,omitempty"`
	EmployeeID   *uint          `gorm:"index" json:"employee_id,omitempty"`
	Employee     *Employee      `gorm:"foreignKey:EmployeeID;references:ID" json:"employee,omitempty"`
	Role         string         `gorm:"type:varchar(20);index" json:"role,omitempty"`
	Action       string         `gorm:"type:varchar(255);not null" json:"action"`
	Endpoint     string         `gorm:"type:varchar(255);not null" json:"endpoint"`
	Method       string         `gorm:"type:varchar(10);not null" json:"method"`
	StatusCode   int            `gorm:"not null" json:"status_code"`
	IPAddress    string         `gorm:"type:varchar(45)" json:"ip_address"`
	UserAgent    string         `gorm:"type:text" json:"user_agent"`
	RequestBody  string         `gorm:"type:text" json:"request_body,omitempty"`
	ResponseBody string         `gorm:"type:text" json:"response_body,omitempty"`
	DurationMs   int64          `gorm:"not null" json:"duration_ms"`
	ErrorMessage string         `gorm:"type:text" json:"error_message,omitempty"`
	CreatedAt    time.Time      `gorm:"index" json:"created_at"`
	DeletedAt    gorm.DeletedAt `gorm:"index" json:"-"`
}
